### PR TITLE
Audit use of defaultTargetPlatform

### PR DESCRIPTION
**Pull request by Hixie**
This should not change any actual behaviour (thus no tests have been affected), but it brings our codebase into a more consistent state in terms of whether we use `defaultTargetPlatform`, `Theme.of(context).platform`, or `Platform` to determine platform-specific behaviour.

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/foundation/platform.dart [hunk 1]
```diff
@@ -11,9 +11,9 @@ import '_platform_io.dart'
 /// This is the default value of [ThemeData.platform] (hence the name). Widgets
 /// from the material library should use [Theme.of] to determine the current
 /// platform for styling purposes, rather than using [defaultTargetPlatform].
-/// However, if there is widget behavior that depends on the actual underlying
-/// platform, then depending on [defaultTargetPlatform] makes sense.
-/// [dart.io.Platform.environment] should be used directly only when it's
+/// Widgets and render objects at lower layers that try to emulate the
+/// underlying platform platform can depend on [defaultTargetPlatform] directly.
```
Comment (Piinks) [inline packages/flutter/lib/src/foundation/platform.dart]: platform platform

**Review by Piinks — APPROVED** [review]
LGTM with exception of doc typo below. :)

File: packages/flutter/lib/src/foundation/platform.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/foundation/platform.dart]: fixed

**Comment by Hixie** [comment]
Please feel free to land this when the tree is green.

**Pull request merged**